cli/src/commands/start.rs

```diff
@@ -285,57 +287,30 @@ impl Start {
         // Initialize the runtime.
         Self::runtime().block_on(async move {
             // Error messages.
-            let node_parse_error = || "Failed to parse node arguments";
-            let display_start_error = || "Failed to initialize the display";
-
-            let (shutdown_tx, shutdown_rx) = oneshot::channel();
+            let node_parse_error = || "Failed to start node";
+            let signal_handler = SignalHandler::new();
 
             // Clone the configurations.
-            let mut cli = self.clone();
-            // Parse the network.
-            match cli.network {
-                MainnetV0::ID => {
-                    // Parse the node from the configurations.
-                    let node = cli
-                        .parse_node::<MainnetV0>(shutdown.clone(), shutdown_tx)
-                        .await
-                        .with_context(node_parse_error)?;
-                    // If the display is enabled, render the display.
-                    if !cli.nodisplay {
-                        // Initialize the display.
-                        Display::start(node, log_receiver).with_context(display_start_error)?;
-                    }
-                }
-                TestnetV0::ID => {
-                    // Parse the node from the configurations.
-                    let node = cli
-                        .parse_node::<TestnetV0>(shutdown.clone(), shutdown_tx)
-                        .await
-                        .with_context(node_parse_error)?;
-                    // If the display is enabled, render the display.
-                    if !cli.nodisplay {
-                        // Initialize the display.
-                        Display::start(node, log_receiver).with_context(display_start_error)?;
-                    }
-                }
-                CanaryV0::ID => {
-                    // Parse the node from the configurations.
-                    let node = cli
-                        .parse_node::<CanaryV0>(shutdown.clone(), shutdown_tx)
-                        .await
-                        .with_context(node_parse_error)?;
-                    // If the display is enabled, render the display.
-                    if !cli.nodisplay {
-                        // Initialize the display.
-                        Display::start(node, log_receiver).with_context(display_start_error)?;
-                    }
-                }
+            let mut self_ = self.clone();
+
+            // Parse the node arguments, start it, and block until shutdown.
+            match self_.network {
+                MainnetV0::ID => self_
+                    .parse_node::<MainnetV0>(log_receiver, signal_handler.clone())
```

Comment:
```suggestion
                    .parse_node::<MainnetV0>(log_receiver, signal_handler)
```
applies to the other spots too